File: backend-api/src/main/java/com/trafficfine/dto/PaymentRequest.java
package com.trafficfine.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record PaymentRequest(
        @NotBlank String referenceNumber,
        @NotBlank String categoryCode,
        @NotBlank String paymentMethod,
        @Pattern(regexp = "\\d{4}", message = "cardLastFourDigits must contain exactly four digits")
        String cardLastFourDigits
) {
}

File: android-app/app/src/main/java/com/example/trafficfinesystem/data/PaymentRequest.kt
package com.example.trafficfinesystem.data

import com.google.gson.annotations.SerializedName

data class PaymentRequest(
    @SerializedName("referenceNumber") val referenceNumber: String,
    @SerializedName("categoryCode") val categoryCode: String,
    @SerializedName("paymentMethod") val paymentMethod: String,
    @SerializedName("cardLastFourDigits") val cardLastFourDigits: String? = null
)
